src/app/clusters/ota-requestor/ota-requestor-server.cpp

```diff
@@ -67,14 +71,61 @@ CHIP_ERROR OtaSoftwareUpdateRequestorAttrAccess::Write(const ConcreteDataAttribu
     switch (aPath.mAttributeId)
     {
     case Attributes::DefaultOtaProviders::Id: {
+        return WriteDefaultOtaProviders(aPath, aDecoder);
+    }
+    default:
+        break;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR OtaSoftwareUpdateRequestorAttrAccess::ReadDefaultOtaProviders(AttributeValueEncoder & aEncoder)
+{
+    chip::OTARequestorInterface * requestor = chip::GetRequestorInstance();
+    if (requestor == nullptr)
+    {
+        return CHIP_ERROR_NOT_FOUND;
+    }
+
+    return requestor->GetDefaultOtaProviderList(aEncoder);
+}
+
+CHIP_ERROR OtaSoftwareUpdateRequestorAttrAccess::WriteDefaultOtaProviders(const ConcreteDataAttributePath & aPath,
+                                                                          AttributeValueDecoder & aDecoder)
+{
+    chip::OTARequestorInterface * requestor = chip::GetRequestorInstance();
+    if (requestor == nullptr)
+    {
+        return CHIP_ERROR_NOT_FOUND;
+    }
+
+    switch (aPath.mListOp)
+    {
+    case ConcreteDataAttributePath::ListOperation::ReplaceAll: {
         DataModel::DecodableList<OtaSoftwareUpdateRequestor::Structs::ProviderLocation::DecodableType> list;
         ReturnErrorOnFailure(aDecoder.Decode(list));
-        // Ignore the list for now
-        break;
+
+        // With chunking, a single large list is converted to a list of AttributeDataIBs. The first AttributeDataIB contains an
+        // empty list (to signal this is a replace so clear out contents) followed by a succession of single AttributeDataIBs for
+        // each entry to be added.
+        size_t count = 0;
+        ReturnErrorOnFailure(list.ComputeSize(&count));
+        VerifyOrReturnError(count == 0, CHIP_ERROR_INVALID_ARGUMENT);
+        return requestor->ClearDefaultOtaProviderList(aDecoder.AccessingFabricIndex());
+    }
+    case ConcreteDataAttributePath::ListOperation::ReplaceItem:
+        return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+    case ConcreteDataAttributePath::ListOperation::DeleteItem:
+        return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+    case ConcreteDataAttributePath::ListOperation::AppendItem: {
+        OtaSoftwareUpdateRequestor::Structs::ProviderLocation::DecodableType item;
+        ReturnErrorOnFailure(aDecoder.Decode(item));
+        return requestor->AddDefaultOtaProvider(aDecoder.AccessingFabricIndex(), item);
```

Comment:
`item` will also have a fabric index, set to `aDecoder.AccessingFabricIndex()`, right?  So could just pass `item` here....

Reply:
I didn't want to make the assumption that IM layer would make sure `item.GetFabricIndex()` is correct (set to the current access fabric index). If I could make that assumption, then yes, just passing in `item` would suffice.

Reply:
You can absolutely make that assumption.

Reply:
I'll pass in only `item` then